### PR TITLE
18526 copy ppms models to engine

### PR DESCRIPTION
**Pull request by darkbushido**
## Description of change

This creates a new rails engine FacilitiesApi
I've modified the generated files so they are able to work in the scope of the engine
Copying over the Provider and Specialties models as well as inverting the version naming convention (v1/ppms instead of ppms/v1)

## Original issue(s)
department-of-veterans-affairs/va.gov-team#11608
department-of-veterans-affairs/va.gov-team#18526

## Things to know about this PR
This is a small building block to puting facilities in an engine.

### CODE REVIEW

File: modules/facilities_api/spec/factories/facilities_api/v1/ppms/specialties.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+FactoryBot.define do
+  factory :v1_ppms_specialty, class: 'V1::Ppms::Specialty' do
```
Comment (LindseySaari) [inline modules/facilities_api/spec/factories/facilities_api/v1/ppms/specialties.rb:4]: Should Ppms be PPMS?

File: config/routes.rb [hunk 2]
```diff
@@ -361,6 +361,7 @@
   # Modules
   mount CovidResearch::Engine, at: '/covid-research'
   mount CovidVaccine::Engine, at: '/covid_vaccine'
+  # mount FacilitiesApi::Engine, at: '/facilities_api'
```
Comment (LindseySaari) [inline config/routes.rb:364]: Are you commenting this out until it's ready to go?

File: modules/facilities_api/README.rdoc [hunk 3]
```diff
@@ -0,0 +1,10 @@
+= FacilitiesApi
+TODO: Short description and motivation.
+
+== Installation
+Ensure the following line is in the root project's Gemfile:
+
+  gem 'facilitiesapi', path: 'modules/facilitiesapi'
```
Comment (LindseySaari) [inline modules/facilities_api/README.rdoc:7]: Was this auto generated? Seems like this should be facilities_api (underscore)

Comment (darkbushido) [reply modules/facilities_api/README.rdoc:7]: It's auto-generated. I had to fix this in several other places but the generator is dropping the `_`

**Review by LindseySaari — APPROVED** [review]

**Pull request merged**